### ISSUE TITLE
Multiple duplicate attached artifact added into promotable-artifacts.properties

### ISSUE THREAD

**Issue by ramvirsingh** [mjiderhamn/promote-maven-plugin#10]
When trying to promote ejb module that also generated a client . promotable-artifacts.properties haas multiple entry for one client.

**Comment by mjiderhamn** [comment]
Hi @ramvirsingh ! Are you able to provide an example of this?

**Comment by ramvirsingh** [comment]
This is my Pom. It should generate ejb modole and client .BUt i have multiple entres for client as in attached "promotable-artifacts.properties"

"POM.xml: [https://github.com/ramvirsingh/SampleEJBAPP/blob/master/pom.xml](url)



promotable-artifacts.properties: [https://github.com/ramvirsingh/SampleEJBAPP/blob/master/target/promotable-artifacts.properties](url)
***************************************
#Generated by promote-maven-plugin
#Tue Feb 13 09:56:28 IST 2018
artifact.file=SampleEJBApp-0.0.1-SNAPSHOT.jar
attached.0.type=ejb-client
artifact.groupId=org.test
attached.2.artifactId=SampleEJBApp
attached.2.id=org.test\:SampleEJBApp\:ejb-client\:client\:0.0.1-SNAPSHOT
attached.2.classifier=client
attached.1.baseVersion=0.0.1-SNAPSHOT
attached.1.type=ejb-client
attached.1.version=0.0.1-SNAPSHOT
artifact.id=org.test\:SampleEJBApp\:ejb\:0.0.1-SNAPSHOT
attached.2.type=ejb-client
attached.2.baseVersion=0.0.1-SNAPSHOT
artifact.type=ejb
attached.1.groupId=org.test
attached.2.version=0.0.1-SNAPSHOT
artifact.artifactId=SampleEJBApp
attached.0.file=SampleEJBApp-0.0.1-SNAPSHOT-client.jar
attached.0.version=0.0.1-SNAPSHOT
attached.0.artifactId=SampleEJBApp
attached.0.classifier=client
attached.0.id=org.test\:SampleEJBApp\:ejb-client\:client\:0.0.1-SNAPSHOT
attached.1.file=SampleEJBApp-0.0.1-SNAPSHOT-client.jar
artifact.version=0.0.1-SNAPSHOT
attached.2.groupId=org.test
attached.0.groupId=org.test
attached.1.artifactId=SampleEJBApp
artifact.baseVersion=0.0.1-SNAPSHOT
attached.2.file=SampleEJBApp-0.0.1-SNAPSHOT-client.jar
attached.1.classifier=client
attached.1.id=org.test\:SampleEJBApp\:ejb-client\:client\:0.0.1-SNAPSHOT
attached.0.baseVersion=0.0.1-SNAPSHOT


Can you please help me out with this. Ideally only one entry for the client also.

**Comment by ramvirsingh** [comment]
This is sample application in github [https://github.com/ramvirsingh/SampleEJBAPP](url)

**Comment by ramvirsingh** [comment]
hi, @mjiderhamn can you help me?

**Comment by mjiderhamn** [comment]
I'm not sure when I will find the time to look at this.

**Comment by mjiderhamn** [comment]
Hello @ramvirsingh . I have **finally** found time to look at this. It seems to me that the problem occurs only when when the artifacts have been built without Maven `clean` goal.

For me
```
mvn clean package
mvn release:clean promote:prepare release:prepare
```
seems to work perfectly fine. Can you please confirm?

I cannot quite remember why I changed the `promotable-artifacts.properties` contents to be "incremental", but I'll look into taking duplicates into account.

**Comment by mjiderhamn** [comment]
That was an easy fix, version 2.0.1 should be on it's way to Maven Central now.
